test/zdtm/lib/test.c

```diff
@@ -238,34 +241,36 @@ void test_init(int argc, char **argv)
 		exit(1);
 	}
 
-	val = getenv("ZDTM_GROUPS");
-	if (val) {
-		char *tok = NULL;
-		unsigned int size = 0, groups[NGROUPS_MAX];
-
-		tok = strtok(val, " ");
-		while (tok) {
-			size++;
-			groups[size - 1] = atoi(tok);
-			tok = strtok(NULL, " ");
+	if (!uid) {
+		val = getenv("ZDTM_GROUPS");
+		if (val) {
+			char *tok = NULL;
+			unsigned int size = 0, groups[NGROUPS_MAX];
```

Comment:
nit: `NGROUPS_MAX` is `65536`
Isn't it will be better to allocate memory from heap?

Reply:
This is just me moving code behind an `if` if running as non-root. I have not changed the actual code. If this should be changed it should happen in another PR.

Reply:
> This is just me moving code behind an `if` if running as non-root. I have not changed the actual code. If this should be changed it should happen in another PR.

no problem. It's just comment.